io-backend/src/main/java/agh/io/iobackend/controller/GameRoomController.java

```diff
@@ -75,56 +85,96 @@ public ResponseEntity<GameRoomResponse> getRoomDetails(@PathVariable Long id) th
     }
 
     @CrossOrigin
-    @GetMapping("/{id}/game") // zwraca id gry - aktualnie to jest nadal roomId
+    @GetMapping("/{id}/game") // zaczecie gry przez Game Mastera - zwraca id gry
     public ResponseEntity<Long> createGame(@PathVariable Long id) throws GameRoomNotFoundException {
-        GameRoom gameRoom = gameRoomService.getGameRoom(id);
-        gameService.createGame(gameRoom);
+        GameRoom gameRoom;
+        try {
+            gameRoom = gameRoomService.getGameRoom(id);
+        } catch (GameRoomNotFoundException e) {
+            return ResponseEntity.badRequest().body(null);
+        }
         gameRoom.setGameStarted(true);
+        Game game = new Game(id, gameRoom.getGameMap());
+        Game savedGame = gameService.createGame(game);
+        gameRoom.setGame(savedGame);
 
-        // szkic pozniejszych zmian z nowa klasa Game podobna do GameRoom
-//        Game game = new Game(id, gameRoom.getMapID(), gameRoom.getGameMasterID());
-//        Game savedGame = gameService.createGame(game);
-//        return ResponseEntity.ok(savedGame.getGameId());
-
-        return ResponseEntity.ok(id);
-
+        return ResponseEntity.ok(savedGame.getGameId());
     }
 
     @CrossOrigin
     @DeleteMapping("/{id}") // room id
-    public ResponseEntity<String> deleteRoom(@PathVariable Long id) throws GameRoomNotFoundException {
-        gameRoomService.deleteGameRoom(id);
-        gameService.deleteGame(id);
+    public ResponseEntity<String> deleteRoom(@PathVariable Long id) {
+        try {
+            gameRoomService.deleteGameRoom(id);
+        } catch (GameRoomNotFoundException e) {
+            return ResponseEntity.badRequest().body("Game room not found");
+        }
+        // cos jeszcze?
```

Comment:
Wydaje mi się że zanim game room usuniemy to trzeba jeszcze Game, bo GameRoom ma na niego referencję